sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableProvider.java

```diff
@@ -36,26 +38,47 @@
 public interface TableProvider {
 
   /**
-   * Return a {@link Table} instance to do read/write with user-specified options.
+   * Return a {@link Table} instance with specified table properties to do read/write.
+   * Implementations should infer the table schema and partitioning.
+   *
+   * @param properties The specified table properties. It's case preserving (contains exactly what
```

Comment:
I did miss most of the discussions unfortunately. Would this be used as table properties directly? Also include read/write options? Or are the read/write options going to translate to a catalog and identifier as we had discussed some time ago?

I guess a path based table would have a `location` table property, which constitutes the old option "path"`, correct?

Reply:
Yes, it's used as table properties directly, not read/write options. The read/write options are case-insensitive and passed through `Table.newScanBuilder`/`Table.newWriteBuilder`.

Table properties doesn't have to be case-insensitive. So here I just define it as case-preserving. The implementation is free to interpret it case-sensitive or case-insensitive. (Spark can't control it anyway)

If you read a table with `DataFrameReader`, then the options will be passed to the data source twice: once with `TableProvider.getTable`, once with `Table.newScanOptions`.

Reply:
a path based table would have a `location` table property, but I don't know why we can't reuse the old name `path`. cc @rdblue